### PR TITLE
Math Magicians: Tests

### PR DESCRIPTION
**Pull request by HafizMassamTabraizKhan**
In this branch, we did the following:

👌  Set up React Testing Library.
🙆‍♂️  Write unit tests for the files operate.js and calculate.js using Jest.
🙆  Create unit tests for all React components.
🙆‍♀️  Use Jest snapshots to test the components.
🆗  Use React Testing Library to simulate user interaction.

### CODE REVIEW

File: src/test/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import calculate from '../logic/calculate';
+
+describe('to test calculate logic', () => {
+  test('calculate logic', () => {
+    expect(calculate({}, 'AC')).toEqual({ total: null, next: null, operation: null });
+  });
+  test('calculate logic', () => {
+    expect(calculate({ total: '1', next: '2', operation: '+' }, '=')).toEqual({ total: '3', next: null, operation: null });
+  });
+});
```
Comment (cynthiainga) [inline src/test/calculate.test.js]: - [x] Good job so far, but as mentioned in the project requirements all the functionalities & scenarios of the calculate.js should be tested. The `.` button should be tested and all the scenarios, the `+/-` button should be tested, and all the scenarios. Please kindly fix that 🙏🏾

File: src/logic/operate.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ export default function operate(numberOne, numberTwo, operation) {
     try {
       return one.mod(two).toString();
     } catch (err) {
-      return "Can't find modulo as can't divide by 0.";
+      return "Can't find module as can't divide by 0.";
```
Comment (cynthiainga) [inline src/logic/operate.js]: - [x] I noticed you make some changes here but the `%` button is named `modulo`,  in definition the `modulo` is the remainder after dividing one number by another, and `module` is one of the separate units that a course of study has been divided into.

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @HafizMassamTabraizKhan, @yashodhicy & @mosidrum, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] Correct Gitflow ✔️ 
- [ ] Used Jest snapshots ✔️ 
- [ ] Good commit history ✔️ 

## Required Changes ♻️

- [ ] Please kindly make sure all your tests are passing, when I run all the tests one doesn’t pass.

Screenshot 2023-06-08 at 13 46 10

Screenshot 2023-06-08 at 13 44 43

- [ ] Please kindly check the comments under the review 👇🏾 👇🏾 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

You can also consider the following:
- I think it would be nice to add the contact info of other contributors in your README file as they contribute to the project 😁 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can get the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/logic/operate.js (hunk 2 above)
Comment (yashodhicy) [reply src/logic/operate.js]: thank you

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @HafizMassamTabraizKhan, @mosidrum and @yashodhicy , 

To highlight 👍🏽 
✔️ Tests pass
✔️ Errors are fixed

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**